backtester/funding/funding.go

```diff
@@ -110,12 +162,46 @@ func (f *FundManager) AddUSDTrackingData(k *kline.DataFromKline) error {
 		if baseSet && quoteSet {
 			return nil
 		}
+		if f.items[i].asset.IsFutures() && k.Item.Asset.IsFutures() {
+			if f.items[i].isCollateral {
+				usdCandles := gctkline.Item{
+					Exchange: k.Item.Exchange,
+					Pair:     currency.Pair{Delimiter: k.Item.Pair.Delimiter, Base: f.items[i].currency, Quote: currency.USD},
+					Asset:    k.Item.Asset,
+					Interval: k.Item.Interval,
+					Candles:  make([]gctkline.Candle, len(k.Item.Candles)),
+				}
+				copy(usdCandles.Candles, k.Item.Candles)
```

Comment:
Why copy if you are going to just adjust everything to 1? Volume differences? Also, the kline item and filtering is done below as well we could probably abstract that out.